core/src/test/java/com/backblaze/b2/json/B2JsonTest.java

```diff
@@ -2792,6 +2793,104 @@ public void testClassWithLotsOfFields() throws IOException, B2JsonException {
+     * close to half of MAX_ARRAY_SIZE.
+     *
+     * Instead, create a subclass of B2JsonByteArrayOutputStream with lowered
+     * MAX_ARRAY_SIZE (1000) for testing purpose: IOException will then be thrown
+     */
+    private static class B2JsonByteArrayOutputStreamForTest extends B2JsonByteArrayOutputStream {
+        private static final int MAX_ARRAY_SIZE = 1000;
+
+        @Override
+        protected void grow(int minCapacity) throws IOException {
+            // overflow-conscious code
+            int oldCapacity = buf.length;
+            int newCapacity = oldCapacity << 1;
+            if (newCapacity - minCapacity < 0)
+                newCapacity = minCapacity;
+            if (newCapacity - MAX_ARRAY_SIZE > 0) {
+                throw new IOException("Requested array size exceeds maximum limit");
+            }
+            buf = Arrays.copyOf(buf, newCapacity);
+        }
+    }
+
+    /* convenience object for testing IOException("Requested array size exceeds maximum limit") */
+    private static class ObjectWithSomeName {
+       @B2Json.required
+       private final String name;
+
+       @B2Json.constructor(params = "name")
+       public ObjectWithSomeName(String name) {
+           this.name = name;
+       }
+
+       @Override
+       public boolean equals(Object o) {
+           if (this == o) {
+               return true;
+           }
+           if (o == null || getClass() != o.getClass()) {
+               return false;
+           }
+           final ObjectWithSomeName objectWithSomeName = (ObjectWithSomeName) o;
+           return this.name.equals(objectWithSomeName.name);
+       }
+
+       @Override
+       public int hashCode() {
+           return name.hashCode();
+       }
+    }
+
+    @Test
+    public void testObjectWithNameToJson() throws B2JsonException, IOException {
+        // an object not big enough to cause IOException("Requested array size exceeds maximum limit")
+        final StringBuilder smallName = new StringBuilder();
+        for (int i = 0; i < 19; i++) {
+            for (char j = 'A'; j <= 'Z'; j++) {
+                smallName.append(j);
+            }
+        }
+        // the total size of object in bytes is 494; doubling will not cause overflow over 1000
+        final ObjectWithSomeName expectedObjectWithSomeSmallName = new ObjectWithSomeName(smallName.toString());
+        final B2JsonByteArrayOutputStreamForTest b2JsonByteArrayOutputStreamForSmall = new B2JsonByteArrayOutputStreamForTest();
+        B2Json.get().toJson(expectedObjectWithSomeSmallName, b2JsonByteArrayOutputStreamForSmall);
+
+        // rebuild the object based on the Json string
+        final String actualJson = b2JsonByteArrayOutputStreamForSmall.toString();
+        final ObjectWithSomeName actualObjectWithSomeName = B2Json.fromJsonOrThrowRuntime(actualJson, ObjectWithSomeName.class);
+        assertEquals(expectedObjectWithSomeSmallName, actualObjectWithSomeName);
```

Comment:
btw, if we get rid of this, we can probably also get rid of equals() and hashCode() on the test class.

Reply:
yes